velox/common/memory/MemoryArbitrator.cpp

```diff
@@ -187,20 +187,41 @@ uint64_t MemoryReclaimer::reclaim(MemoryPool* pool, uint64_t targetBytes) {
   if (pool->kind() == MemoryPool::Kind::kLeaf) {
     return 0;
   }
-  // TODO: add to sort the child memory pools based on the reclaimable bytes
-  // before memory reclamation.
+
+  // Sort the child pools based on their reserved memory and reclaim from the
```

Comment:
It feels that we may want to iterate on this logic and maybe customize it based on application. I wonder if it makes sense to extract this logic into a separate class. Also, the TODO said to sort by "reclaimable bytes", not "reserved bytes". Reclaiming from the operator with most bytes available to reclaim makes more sense, no?

Reply:
The reserved bytes is more relevant here because we can also release unused reservation at leaf level. My previous version adds a ordered reclaim flag in memory reclaimer to customize the logic.